protocols/mdns/src/behaviour.rs

```diff
@@ -155,6 +152,13 @@ impl Mdns {
             Async::new(socket)?
         };
         let if_watch = if_watch::IfWatcher::new().await?;
+        // randomize timer to prevent all converging and firing at the same time.
+        let query_interval = {
+            use rand::Rng;
+            let mut rng = rand::thread_rng();
+            let jitter = rng.gen_range(0..100);
+            config.query_interval + Duration::from_millis(jitter)
```

Comment:
Why increasing the interval itself instead of offsetting the initial trigger only and continuing on a fixed interval from there on? Wouldn't that suffice?

Reply:
They converge to firing at the same time if the interval is the same for all nodes, due to the conditions that cause the timers to reset being the same. We can add randomness every time we reset the timer or just give nodes a slightly different interval once. I don't think it makes a big difference, since fairness is not required. It doesn't matter if the same node always wins.

Reply:
I missed that `self.query_interval` instead of `config.query_interval` is used from here on.  Thanks for the clarification!